drivers/flash/flash_mcux_flexspi_mx25um51345g.c

```diff
@@ -395,8 +395,29 @@ static int flash_flexspi_nor_write(const struct device *dev, off_t offset,
 		 * be wrapped around within the same page
 		 */
 		i = MIN(SPI_NOR_PAGE_SIZE - (offset % SPI_NOR_PAGE_SIZE), len);
+		/* Save off the amount of data to be written and the end position before the offset
+		 * and i values are changed for alignment in the OPI DTR case.
+		 */
+		size_t processed_length = i;
+		off_t data_end = offset + processed_length;
 #ifdef CONFIG_FLASH_MCUX_FLEXSPI_NOR_WRITE_BUFFER
-		memcpy(nor_write_buf, src, i);
+		uint8_t *data_start = nor_write_buf;
+#ifdef CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G_OPI_DTR
+		/* Check if start address is even and pad one byte before the data if it is not. */
+		if (offset % 2) {
```

Comment:
Probably compiler will optimize this anyway into bit check, but it is more efficient to just check the LSb to figure out whether number is even or odd. Some architecture have bit-test-and-branch operations that work well in such situations.

Reply:
good point here- I'll update PR with this change